conformance/utils/http/http.go

```diff
@@ -28,19 +28,33 @@ import (
 
 // ExpectedResponse defines the response expected for a given request.
 type ExpectedResponse struct {
-	Request    ExpectedRequest
+	// Request defines the request to make.
+	Request Request
+
+	// BackendRequest defines the request that
+	// is expected to arrive at the backend. If
+	// not specified, it is assumed to be the
+	// same as the original request.
+	BackendRequest *Request
```

Comment:
This could have been reversed so that the "make request" code isn't dealing with a struct that includes AbsentHeaders

To do that, ExpectedRequest  would embed Request, and adding AbsentHeaders. You'd always supply an ExpectedRequest, and it would be used for making the request if no Request was specified

But ... meh

Reply:
Agree that this structure would be a bit better if it's not too much trouble.

Reply:
OK, I can make this change, just noting that this makes all of the struct literals more verbose in the test cases due to the additional nesting (which I think is why I decided against this approach originally).

Reply:
Will push a commit shortly that I think addresses the essence of this comment while keeping the test case struct literals simple for the normal cases.

Reply:
Alright, the latest version splits out two structs, `Request` and `ExpectedRequest`, with the latter embedding the former and adding `AbsentHeaders`. `Request` is the required one, though, with `ExpectedRequest` defaulting to matching `Request` if not specified. I think this is the best of both worlds, PTAL.